gravitl/netmaker
# Story/gra 1252

## Describe your changes

This modifies the nm-quick script to do the following:
1. Adds a "-a" flag, which will run "auto install" with default options. No prompts will be used.
2. Installs the netclient into the docker-compose, makes it a default host and an ingress gw

The nm-quick script was modified to use functions, rather than one long script, which is why the changes look so big, but it is mostly just reformatting the existing script.

This also makes a small change to nm-upgrade to replace some faulty logic, and adds an example netclient docker-compose file to the compose folder.

## Provide Issue ticket number if applicable/not in title

GRA-1252 (this is one part of a bigger ticket)

## Provide testing steps

1. Run "./nm-quick.sh -a" on a fresh linux machine
2. See that netmaker successfully installs, and there is a healthy netclient running

## Checklist before requesting a review
